test/linalg/schur.jl

```diff
@@ -20,69 +20,80 @@ for eltya in (Float32, Float64, Complex64, Complex128, Int)
     a = eltya == Int ? rand(1:7, n, n) : convert(Matrix{eltya}, eltya <: Complex ? complex(areal, aimg) : areal)
     asym = a'+a                  # symmetric indefinite
     apd  = a'*a                 # symmetric positive-definite
-    ε = εa = eps(abs(float(one(eltya))))
+    for atype in ("Array", "SubArray")
+        if atype == "Array"
+            a = a
+        else
+            a = sub(a, 1:n, 1:n)
+            asym = sub(asym, 1:n, 1:n)
+            apd = sub(apd, 1:n, 1:n)
+        end
+        ε = εa = eps(abs(float(one(eltya))))
 
-debug && println("\ntype of a: ", eltya, "\n")
-debug && println("Schur")
-    d,v = eig(a)
-    f   = schurfact(a)
-    @test_approx_eq f[:vectors]*f[:Schur]*f[:vectors]' a
-    @test_approx_eq sort(real(f[:values])) sort(real(d))
-    @test_approx_eq sort(imag(f[:values])) sort(imag(d))
-    @test istriu(f[:Schur]) || eltype(a)<:Real
-    @test_approx_eq full(f) a
-    @test_throws KeyError f[:A]
+    debug && println("\ntype of a: ", eltya, "\n")
+    debug && println("Schur")
+        d,v = eig(a)
+        f   = schurfact(a)
+        @test_approx_eq f[:vectors]*f[:Schur]*f[:vectors]' a
+        @test_approx_eq sort(real(f[:values])) sort(real(d))
+        @test_approx_eq sort(imag(f[:values])) sort(imag(d))
+        @test istriu(f[:Schur]) || eltype(a)<:Real
+        @test_approx_eq full(f) a
+        @test_throws KeyError f[:A]
 
-debug && println("Reorder Schur")
-    # use asym for real schur to enforce tridiag structure
-    # avoiding partly selection of conj. eigenvalues
-    ordschura = eltya <: Complex ? a : asym
-    S = schurfact(ordschura)
-    select = bitrand(n)
-    O = ordschur(S, select)
-    sum(select) != 0 && @test_approx_eq S[:values][find(select)] O[:values][1:sum(select)]
-    @test_approx_eq O[:vectors]*O[:Schur]*O[:vectors]' ordschura
-    @test_throws KeyError f[:A]
-    Snew = Base.LinAlg.Schur(copy(S.T), copy(S.Z), copy(S.values))
-    SchurNew = ordschur!(Snew, select)
-    @test O[:vectors] ≈ SchurNew[:vectors]
-    @test O[:Schur] ≈ SchurNew[:Schur]
+    debug && println("Reorder Schur")
+        # use asym for real schur to enforce tridiag structure
+        # avoiding partly selection of conj. eigenvalues
+        ordschura = eltya <: Complex ? a : asym
+        S = schurfact(ordschura)
+        select = bitrand(n)
+        O = ordschur(S, select)
+        sum(select) != 0 && @test_approx_eq S[:values][find(select)] O[:values][1:sum(select)]
+        @test_approx_eq O[:vectors]*O[:Schur]*O[:vectors]' ordschura
+        @test_throws KeyError f[:A]
+        Snew = Base.LinAlg.Schur(copy(S.T), copy(S.Z), copy(S.values))
```

Comment:
I had a suspicion that this would be the result. Would you mind adding the two `copy` methods for `Schur` and `GeneralizedSchur` to this PR? They should be almost trivial.